npm/private/npm_link_package.bzl

```diff
@@ -292,25 +291,29 @@ deps of npm_link_package_store must be in the same package.""" % (ctx.label.pack
                             dep_ref_dep_symlink_path = paths.join("node_modules", utils.virtual_store_root, dep_virtual_store_name, "node_modules", dep_ref_dep_alias)
                             transitive_files.extend(_make_symlink(ctx, dep_ref_dep_symlink_path, dep_ref_def_virtual_store_directory))
 
-    direct_files = depset(direct = direct_files)
-    transitive_files = depset(direct = transitive_files)
-    transitive_closure_depsets = [direct_files, transitive_files]
+    files = [virtual_store_directory] if virtual_store_directory else []
+    files_depset = depset(files)
+
+    transitive_files_depsets = [depset(direct = transitive_files)]
     transitive_declarations_depsets = []
 
     for dep in ctx.attr.deps:
-        transitive_closure_depsets.append(dep[DefaultInfo].files)
+        transitive_files_depsets.append(dep[NpmLinkedPackageStoreInfo].transitive_files)
         if DeclarationInfo in dep:
             transitive_declarations_depsets.append(dep[DeclarationInfo].transitive_declarations)
 
-    result = [
-        DefaultInfo(files = depset(transitive = transitive_closure_depsets)),
+    transitive_files_depset = depset(files, transitive = transitive_files_depsets)
+    transitive_declarations_depset = depset(files, transitive = transitive_files_depsets + transitive_declarations_depsets)
+
+    providers = [
+        DefaultInfo(files = transitive_files_depset),
         # Always assume that packages provide typings, so we don't need to use an action to
         # inspect the package.json#typings field or search for .d.ts files in the package and
         # so that any package can be passed as dep to rules that require DeclarationInfo such
         # as ts_project.
         DeclarationInfo(
-            declarations = direct_files,
-            transitive_declarations = depset(transitive = [transitive_files] + transitive_declarations_depsets),
+            declarations = files_depset,
```

Comment:
Should `declarations` not only be assigned declarations? Why the full `files_depset`?

Reply:
Same question in a few other places...

Reply:
I think I broke GitHub. This diff view is out-of date and showing the wrong filename 🤷 . Guess I have too many commits?

Reply:
Oh, I force pushed the same commits and now it seems to be displaying correctly..

Reply:
This is out of date now that the code has been refactored further